lib/ansible/plugins/lookup/filament_lookup.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: filament_lookup
+    author: Ansible core team
+    version_added: "2.7"
+    short_description: reuturn output of process table.
+    description:
+      - If recieved no arguments, return the whole process table. If recieved arguments, search the process tale with the argument as a process name.
+"""
+
+EXAMPLES = """
+  result: "{{lookup('filament_lookup')}}"
+  tasks:
+  - debug:
+    msg: "result is {{result}}"
+"""
+
+RETURN = """
+content of process table
+"""
+
+
+from ansible.plugins.lookup import LookupBase
+
+try:
+    import subprocess
```

Comment:
Oh, this will cause fallback display? I just see other plugins do this and I copied. What's the right way to import display?

Reply:
The problem I'm trying to spot is that you have wrapped a number of imports here, so if **any** of them fails:
1) next import lines **won't be executed at all**
2) control flow will jump into except block, so that you'll have other import which will be executed and then `display` var initialized

After that, all other code in module will get executed and it seems to rely on the fact that all of other imports succeeded.
In case if they failed you'll get a `NameError`, which will state that it doesn't know about `AnsibleError`, for example, and it doesn't exist.

That workaround is designed to provide a fallback for `display` if importing it from `__main__` module fails.
But by wrapping all the imports you're simply masking lots of different errors, which would have to be processed separately.

So the rule of thumb would be to narrow down try block to some concrete piece of code, which you expect to raise an exception. It is considered harmful to "swallow" arbitrary errors without proper handling of corresponding situations.